agent/hibinit-agent

```diff
@@ -29,53 +29,53 @@ except:
 
 # Space reserved for swap headers
 SWAP_RESERVED_SIZE = 16384
-log_to_syslog = True
+LOG_TO_SYSLOG = True
+
 SWAP_FILE = '/swap'
 
 DEFAULT_STATE_DIR = '/var/lib/hibinit-agent'
 HIB_ENABLED_FILE = "hibernation-enabled"
 IMDS_BASEURL = 'http://169.254.169.254'
 IMDS_API_TOKEN_PATH = 'latest/api/token'
 IMDS_SPOT_ACTION_PATH = 'latest/meta-data/hibernation/configured'
-MAX_SWAP_SIZE_OFFSET_ALLOWED = 100 * 1024
+MAX_SWAP_SIZE_OFFSET_ALLOWED = 100 * 1024  # 100 KiB
 
 # Sigterm_handler Behaviour Modifiers
-CRITICAL_PROCESS_IN_PROGRESS = False
 SHUTDOWN_REQUESTED = False
 
 
-def log(message):
-    if log_to_syslog:
+def print_to_sys_log(message):
+    if LOG_TO_SYSLOG:
         syslog.syslog(message)
 
 
-def sigterm_handler(signum, frame):
-    if CRITICAL_PROCESS_IN_PROGRESS:
-        global SHUTDOWN_REQUESTED
-        SHUTDOWN_REQUESTED = True
-    else:
-        cmd = "swapon -s | cut -f1,1 | grep -w {name}"
-        cmd = cmd.format(name=SWAP_FILE)
-        swapoff = "swapoff {filename}"
-        cmd = check_output(cmd, shell=True)
-        if cmd.strip() == SWAP_FILE:
-            swapoff = swapoff.format(filename=SWAP_FILE)
-            check_call(swapoff, shell=True)
-        if os.path.isfile(SWAP_FILE) and os.access(SWAP_FILE, os.R_OK):
-            os.remove(SWAP_FILE)
-        exit(0)
+def critical_process_sigterm_handler(signum, frame):
+    global SHUTDOWN_REQUESTED
+    SHUTDOWN_REQUESTED = True
 
 
-def begin_critical_process():
-    global CRITICAL_PROCESS_IN_PROGRESS
-    CRITICAL_PROCESS_IN_PROGRESS = True
+def default_sigterm_handler(signum, frame):
+    check_swapon_cmd = "swapon -s | cut -f1,1 | grep -w {name}"
+    check_swapon_cmd = check_swapon_cmd.format(name=SWAP_FILE)
+    check_swapon = check_output(check_swapon_cmd, shell=True)
+    if check_swapon.strip() == SWAP_FILE:
+        swapoff_cmd = "swapoff {filename}"
+        swapoff_cmd = swapoff_cmd.format(filename=SWAP_FILE)
+        check_call(swapoff_cmd, shell=True)
+    if os.path.isfile(SWAP_FILE) and os.access(SWAP_FILE, os.R_OK):
+        os.remove(SWAP_FILE)
+    exit(0)
 
 
-def end_critical_process():
-    global CRITICAL_PROCESS_IN_PROGRESS
-    CRITICAL_PROCESS_IN_PROGRESS = False
-    if SHUTDOWN_REQUESTED:
-        exit(0)
+def critical_process(function_with_critical_process):
+    def wrapper(*args, **kwargs):
```

Comment:
Why a wrapper function is required ?

Reply:
Further details explained in change log `Refactored critical_process graceful termination logic using a decorator @critical_process`

If the question is about the use of the wrapper function to return the transformed function, this is common practice for decorators

Not sure if the question is about decorators in general, but link to [PEP 318](https://peps.python.org/pep-0318/) for more details

```
@critical_process
def funct

funct = critical_process(funct)
```